### PR TITLE
ISSD-1352 [MARKETPLACE] Add Create License Flow Structure 

### PR DESCRIPTION
**Pull request by joanagrego**

### CODE REVIEW

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/FooterButtons/index.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		backButton: ButtonProps;
+		customizedButton: ButtonProps;
+		nextButton: ButtonProps;
+	};
+	onClickCancel?: () => void;
+	onClickCustomized: () => void;
+	onClickNext: () => void;
+};
+
+export function FooterButtons({
```
Comment (silasbsi) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/FooterButtons/index.tsx:29]: Please, use a arrow function on this component.

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/RadioCardList/components/RadioCard.tsx [hunk 2]
```diff
@@ -34,7 +40,7 @@ const NewRadioCard = ({
 	return (
 		<div
 			className={classNames(
-				'align-items-center cursor-pointer d-flex form-control justify-content-between mb-5 px-0 py-4 radio-card',
+				`align-items-center cursor-pointer d-flex form-control justify-content-between mb-5 px-0 py-4 radio-card`,
```
Comment (silasbsi) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/RadioCardList/components/RadioCard.tsx:43]: You don't need to use `` like used on the template literals here.
```suggestion
				'align-items-center cursor-pointer d-flex form-control justify-content-between mb-5 px-0 py-4 radio-card',
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx [hunk 3]
```diff
@@ -56,6 +60,8 @@ const GetAppFlow = () => {
 	const [orderType, setOrderType] = useState<OrderType>();
 	const [purchaseOrderNumber, setPurchaseOrderNumber] = useState<string>('');
 	const [step, setStep] = useState<StepType>(StepType.ACCOUNT);
+	const [hasTrial, setHasTrial] = useState(false);
+	const [basePrice, setBasePrice] = useState(0);
```
Comment (silasbsi) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx:64]: Please, add the type here.

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx [hunk 4]
```diff
@@ -239,15 +245,111 @@ const GetAppFlow = () => {
 		},
 	};
 
+	const getProductBasePriceAndTrial = (skus: SKU[]) => {
+		skus?.forEach((sku) => {
+			const licenseUsageTypes = sku?.skuOptions.filter(
```
Comment (silasbsi) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx:250]:
```suggestion
			const licenseUsageTypes = sku?.skuOptions?.filter(
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx [hunk 5]
```diff
@@ -239,15 +245,111 @@ const GetAppFlow = () => {
...
+				(skuOption) =>
+					skuOption?.value === SkuOptions.STANDARD.toLowerCase() ||
+					skuOption?.value === SkuOptions.TRIAL.toLowerCase()
+			);
+
+			licenseUsageTypes.forEach((licenseUsageType) => {
```
Comment (silasbsi) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx:256]:
```suggestion
			licenseUsageTypes?.forEach((licenseUsageType) => {
```

**Review by silasbsi — CHANGES_REQUESTED** [review]
I added some suggestions

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CreateLicense/index.tsx [hunk 6]
```diff
@@ -3,39 +3,186 @@
...
+				<small className="col-6 col-md-4 font-weight-bold m-0">
+					Start Date - Exp. Date
+				</small>
+				<small className="col-6 col-md-4 subscription-banner-text text-nowrap">
+					{ProductCardInfo.licenseKeyData.startDate} &ndash;{' '}
+					{ProductCardInfo.licenseKeyData.endDate}
+				</small>
+			</div>
+		</>
+	);
```
Comment (claricebrasil) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CreateLicense/index.tsx:142]: Can we transform this component in an independent component?

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CreateLicense/index.tsx [hunk 7]
```diff
@@ -3,39 +3,186 @@
...
+					RightSideBanner={() => (
+						<AccountEmailInfo
+							productCreatorAccount={
+								ProductCardInfo.productCreatorAccount
+							}
+							userAccount={ProductCardInfo.userAccount}
+						/>
+					)}
+					creatorAccount={
+						ProductCardInfo.productCreatorAccount as any
```
Comment (claricebrasil) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CreateLicense/index.tsx:177]:
```suggestion
						ProductCardInfo.productCreatorAccount as string
```
Isn't this type a string one?

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/FooterButtons/index.tsx [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+				</ClayButton>
+			)}
+
+			<div className="d-flex justify-content-end">
+				{customizedButton?.show && (
+					<ClayButton
+						displayType={
+							customizedButton?.displayType as DisplayType
+						}
+						onClick={() => onClickCustomized()}
```
Comment (claricebrasil) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/FooterButtons/index.tsx:56]:
```suggestion
						onClick={() => onClickCustomized()}
```
What this function does? Can we use a more descriptive name?

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CreateLicense/index.tsx [hunk 9]
```diff
@@ -3,39 +3,186 @@
...
+					}
+					product={ProductCardInfo.product as any}
```
Comment (claricebrasil) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CreateLicense/index.tsx:179]: Can we add a type for product here?

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx [hunk 10]
```diff
@@ -239,15 +245,111 @@ const GetAppFlow = () => {
...
+				Account Selected
+			</strong>
+
+			<AccountEmailInfo
+				productCreatorAccount={productCreatorAccount}
+				userAccount={myUserAccount}
+			/>
+		</div>
+	);
+
```
Comment (claricebrasil) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx:344]: Can we make this an independent component?

**Review by claricebrasil — APPROVED** [review]
Great job! Just added some suggestions here

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx (hunk 3 above)
Comment (joanagrego) [reply workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx:64]: Done.

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CreateLicense/index.tsx (hunk 6 above)
Comment (joanagrego) [reply workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CreateLicense/index.tsx:142]: Hi @claricebrasil, thanks for the suggestion. The thing is: This is already a component of the Product Card.

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/FooterButtons/index.tsx (hunk 8 above)
Comment (joanagrego) [reply workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/FooterButtons/index.tsx:56]: Done!

**Comment by joanagrego** [comment]
ci:test:sf

**Comment by joanagrego** [comment]
ci:test:sf

**Review by silasbsi — APPROVED** [review]
Good job @joanagrego thanks :martial_arts_uniform:

**Comment by joanagrego** [comment]
ci:forward:force

**Pull request closed without merging**